roles/ec2_instance_create_delete/README.md

```diff
@@ -0,0 +1,171 @@
+# ec2_instance_create
+
+A role to create an EC2 instance in AWS.
+
+Users can specify various parameters for instance configuration, including instance type, AMI ID, key pair, tags, and VPC/subnet configuration.
+
+This role also supports the creation of optional networking resources, such as an external security group and an Elastic IP (EIP). You can choose to wait for the EC2 instance to finish booting before continuing.
+
+## Role Variables
+
+The following variables can be set in the role to customize EC2 instance creation and networking configurations:
+
+### Role Operation
+
+* **ec2_instance_create_delete_operation**: (Optional)
+  - Goal state for the instances.
+  - "O(state=present): ensures instances exist, but does not guarantee any state (e.g. running). Newly-launched instances will be run by EC2."
+  - "O(state=running): O(state=present) + ensures the instances are running."
+  - "O(state=started): O(state=running) + waits for EC2 status checks to report OK if O(wait=true)."
+  - "O(state=stopped): ensures an existing instance is stopped."
+  - "O(state=rebooted): convenience alias for O(state=stopped) immediately followed by O(state=running)."
+  - "O(state=restarted): convenience alias for O(state=stopped) immediately followed by O(state=started)."
+  - "O(state=terminated): ensures an existing instance is terminated."
+  - "O(state=absent): alias for O(state=terminated)."
+  choices are [present, terminated, running, started, stopped, restarted, rebooted, absent]
+  Default is `present`.
+
+### EC2 Instance Configuration
+
+* **ec2_instance_create_delete_aws_region**: (Required)
+  The AWS region in which to create the EC2 instance.
+
+* **ec2_instance_create_delete_instance_name**: (Required)
+  The name of the EC2 instance to be created.
+
+* **ec2_instance_create_delete_instance_type**: (Required)
+  The instance type for the EC2 instance (e.g., `t2.micro`, `m5.large`).
+
+* **ec2_instance_create_delete_ami_id**: (Required)
+  The AMI ID for the EC2 instance.
+
+* **ec2_instance_create_delete_key_name**: (Optional)
+  The name of the key pair to use for SSH access to the EC2 instance.
+  If the key does not exist, a key pair will be created with the name.
+
+* **ec2_instance_create_delete_vpc_subnet_id**: (Optional)
+  The ID of the VPC subnet in which the instance will be launched.
+  If not provided, instance might get created with `default` subnet in the AWS region if present.
+
+* **ec2_instance_create_delete_tags**: (Optional)
+  A dictionary of tags to assign to the EC2 instance.
+
+* **ec2_instance_create_delete_wait_for_boot**: (Optional)
+  Whether to wait for the EC2 instance to be in the "running" or "terminated" state before continuing. Default is `true`.
+
+### Optional Networking Resources
+
+#### Elastic IP
+
+* **ec2_instance_create_delete_vpc_id**: (Optional)
```

Comment:
I'm not entirely sure why this is needed. Shouldn't we just use the VPC of the instance?